### PR TITLE
Add explanation for `lock_obj` to `JournalFileStorage` docstring

### PR DESCRIPTION
**Pull request by eukaryo**
## Motivation

I want to add explanation for `lock_obj`, which is an optional argument of `JournalFileStorage` class.

## Description of the changes

Added explanation for `lock_obj` to `JournalFileStorage` docstring.

### CODE REVIEW

File: optuna/storages/_journal/file.py [hunk 1]
```diff
@@ -171,6 +171,9 @@ class JournalFileStorage(BaseJournalLogStorage):
 
         lock_obj:
             Lock object for process exclusivity.
+
+    A instance of :class:`~optuna.storages.JournalFileSymlinkLock` and
```
Comment (nabenabe0928) [inline optuna/storages/_journal/file.py:175]:
```suggestion
    An instance of :class:`~optuna.storages.JournalFileSymlinkLock` and
```

**Comment by eukaryo** [comment]
The target page will changed as following: https://optuna--5540.org.readthedocs.build/en/5540/reference/generated/optuna.storages.JournalFileStorage.html#optuna.storages.JournalFileStorage

File: optuna/storages/_journal/file.py [hunk 2]
```diff
@@ -171,6 +171,9 @@ class JournalFileStorage(BaseJournalLogStorage):
 
         lock_obj:
             Lock object for process exclusivity.
```
Comment (nabenabe0928) [inline optuna/storages/_journal/file.py:173]: How about moving the newly added explanation to the section of `lock_obj`?

**Review by nabenabe0928 — COMMENTED** [review]
Thank you for the PR, I left two minor comments!

File: optuna/storages/_journal/file.py (hunk 2 above)
Comment (eukaryo) [reply optuna/storages/_journal/file.py]: Agreed. I revised the docs.

**Review by nabenabe0928 — APPROVED** [review]
@eukaryo 
I confirmed your change, LGTM
image

**Review by not522 — APPROVED** [review]
LGTM!

**Pull request merged**